.circleci/set-up-globals.sh

```diff
@@ -0,0 +1,129 @@
+
+
+#=====================================================================================================================
+# Start EXPORTing needed environment variables
+# Circle CI 2.0 does not yet expand environment variables so they have to be manually EXPORTed
+# Once environment variables can be expanded this section can be removed
+# See: https://discuss.circleci.com/t/unclear-how-to-work-with-user-variables-circleci-provided-env-variables/12810/11
+# See: https://discuss.circleci.com/t/environment-variable-expansion-in-working-directory/11322
+# See: https://discuss.circleci.com/t/circle-2-0-global-environment-variables/8681
+#=====================================================================================================================
+echo 'export PATH=$PATH:$HOME/bin:$HOME/terminus/bin' >> $BASH_ENV
+echo 'export BRANCH=$(echo $CIRCLE_BRANCH | grep -v '"'"'^\(master\|[0-9]\+.x\)$'"'"')' >> $BASH_ENV
+echo 'export PR_ENV=${BRANCH:+pr-$BRANCH}' >> $BASH_ENV
+echo 'export CIRCLE_ENV=ci-$CIRCLE_BUILD_NUM' >> $BASH_ENV
+# If we are on a pull request
+if [[ $CIRCLE_BRANCH != "master" && -n ${CIRCLE_PULL_REQUEST+x} ]]
+then
+	# Then use a pr- branch/multidev
+	PR_NUMBER=${CIRCLE_PULL_REQUEST##*/}
+	PR_BRANCH="pr-${PR_NUMBER}"
+	echo "export DEFAULT_ENV=pr-${PR_NUMBER}" >> $BASH_ENV
+else
+	# otherwise make the branch name multidev friendly
+	if [[ $CIRCLE_BRANCH == "master" ]]
+	then
+		echo "export DEFAULT_ENV=dev" >> $BASH_ENV
+	else
+		echo 'export DEFAULT_ENV=$(echo ${PR_ENV:-$CIRCLE_ENV} | tr '"'"'[:upper:]'"'"' '"'"'[:lower:]'"'"' | sed '"'"'s/[^0-9a-z-]//g'"'"' | cut -c -11 | sed '"'"'s/-$//'"'"')' >> $BASH_ENV
+	fi
+fi
+echo 'export TERMINUS_ENV=${TERMINUS_ENV:-$DEFAULT_ENV}' >> $BASH_ENV
+source $BASH_ENV
+#===========================================
+# End EXPORTing needed environment variables
+#===========================================
+
+terminus --version
+
+#===============================
+# Start Install Terminus Plugins
+#===============================
+
+INSTALL_TERMINUS_PLUGINS() {
+	composer create-project -n -d $HOME/.terminus/plugins pantheon-systems/terminus-build-tools-plugin:$BUILD_TOOLS_VERSION
+	composer create-project -n -d $HOME/.terminus/plugins pantheon-systems/terminus-secrets-plugin:^1
+}
+
+# Create Terminus plugins directory and install plugins if needed
+if [ ! -d $HOME/.terminus/plugins ]
+then
+	mkdir -p $HOME/.terminus/plugins
+	INSTALL_TERMINUS_PLUGINS
+fi
+
+# Stash the current time
+CURRENT_TIMESTAMP=$(date +%s)
+if [ ! -f $HOME/.terminus/plugins/last-updated.txt ]
+then
+	echo $CURRENT_TIMESTAMP > $HOME/.terminus/plugins/last-updated.txt
+fi
+
+# Stash the time Terminus plugins were last updated
+TERMINUS_PLUGINS_UPDATED=$(cat $HOME/.terminus/plugins/last-updated.txt)
+
+# Update Terminus plugins if they are more than 24 hours old
+# Otherwise cached version will be used if they exist
+if [ "$CURRRENT_TIMESTAMP - $TERMINUS_PLUGINS_UPDATED" -gt "86400" ]
+then
+	INSTALL_TERMINUS_PLUGINS
```

Comment:
If you did do it like this, I would expect that composer create-project would fail unless you deleted the old directories first. But overall, I think that `git pull` would be a better update mechanism.